homeassistant/components/sensor/openexchangerates.py

```diff
@@ -0,0 +1,90 @@
+"""Support for openexchangerates.org exchange rates service."""
+from datetime import timedelta
+import logging
+import requests
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_API_KEY
+
+_RESOURCE = 'https://openexchangerates.org/api/latest.json'
+_LOGGER = logging.getLogger(__name__)
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=100)
+CONF_BASE = 'base'
+CONF_QUOTE = 'quote'
+CONF_NAME = 'name'
+DEFAULT_NAME = 'Exchange Rate Sensor'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Openexchangerates sensor."""
+    payload = config.get('payload', None)
+    rest = OpenexchangeratesData(
+        _RESOURCE,
+        config.get(CONF_API_KEY),
+        config.get(CONF_BASE, 'USD'),
+        config.get(CONF_QUOTE),
+        payload
+    )
+    rest.update()
```

Comment:
Yep. Just make sure it won't get added when it is an error.

On Tue, Jun 21, 2016 at 8:58 AM, arsaboo <notifications@github.com> wrote:

> In homeassistant/components/sensor/openexchangerates.py
> <https://github.com/home-assistant/home-assistant/pull/2310#discussion_r67897873>
> :
>
> > +CONF_QUOTE = 'quote'
> > +CONF_NAME = 'name'
> > +DEFAULT_NAME = 'Exchange Rate Sensor'
> > +
> > +
> > +def setup_platform(hass, config, add_devices, discovery_info=None):
> > +    """Setup the Openexchangerates sensor."""
> > +    payload = config.get('payload', None)
> > +    rest = OpenexchangeratesData(
> > +        _RESOURCE,
> > +        config.get(CONF_API_KEY),
> > +        config.get(CONF_BASE, 'USD'),
> > +        config.get(CONF_QUOTE),
> > +        payload
> > +    )
> > +    rest.update()
>
> I couldn't find a way to validate the key. However, it returns errors
> <https://docs.openexchangerates.org/docs/errors> in the following format
>
> {
> "error": true,
> "status": 401,
> "message": "invalid_app_id",
> "description": "Invalid App ID provided - please sign up at
> https://openexchangerates.org/signup, or contact
> support@openexchangerates.org."
> }
>
> May be that can be used to validate the API.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/2310/files/c0015e619687222c66a661f310bbcaba44f1ad92#r67897873>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABYJ2iilysoVNpvtSq7TFpOF6TmwiPdUks5qOAo-gaJpZM4I2-nR>
> .
>



-- 
PaulusSchoutsen.nl
It's nice to be important but it's more important to be nice.